src/components/MyDocuments.tsx

```diff
@@ -1,31 +1,251 @@
+              this.setState({
+                buttonState: '',
+                pdfFiles: undefined,
+              }, () => this.getDocumentData());
+            } else {
+              alert.show(`Failure to upload ${pdfFile.name}`);
+              this.setState({ buttonState: '' });
+            }
+          });
+      }
+    } else {
+      alert.show('Please select a file');
+      this.setState({ buttonState: '' });
+    }
+  }
+
+  static maxFilesExceeded(files, maxNumFiles) {
+    return files.length > maxNumFiles;
+  }
+
+  static fileNamesUnique(files) {
+    const fileNames : string[] = [];
+    for (let i = 0; i < files.length; i += 1) {
+      const fileName = files[i].name;
+      fileNames.push(fileName);
+    }
+
+    return fileNames.length === new Set(fileNames).size;
+  }
+
+  handleChangeFileUpload(event: any) {
+    event.preventDefault();
+    const {
+      alert,
+    } = this.props;
+    const { files } = event.target;
+
+    // check that the number of files uploaded doesn't exceed the maximum
+    if (files.length > MAX_NUM_OF_FILES) {
+      // eslint-disable-next-line no-param-reassign
+      event.target.value = null; // discard selected files
+      alert.show(`A maximum of ${MAX_NUM_OF_FILES} files can be uploaded at a time`);
+      return;
+    }
+
+    // all validation met
+    this.setState({
+      pdfFiles: files,
+    });
+  }
+
+  handleChangeFileDownload(event: any, rowIndex: number) {
+    event.preventDefault();
+    const {
+      alert,
+    } = this.props;
+    const { files } = event.target;
+
+    this.setState({
+      pdfFiles: files,
+    }, () => this.handleFileDownload(rowIndex));
+  }
+
+  handleFileDownload(rowIndex: number) {
+    const {
+      userRole,
+    } = this.props;
+
+    const documentId = this.state.documentData[rowIndex].id;
+    const documentName = this.state.documentData[rowIndex].filename;
+
+    let pdfType;
+    if (userRole === Role.Worker || userRole === Role.Admin || userRole === Role.Director) {
+      pdfType = PDFType.APPLICATION;
+    } else if (userRole === Role.Client) {
+      pdfType = PDFType.IDENTIFICATION;
+    } else {
+      pdfType = undefined;
+    }
+
+    fetch(`${getServerURL()}/download`, {
+      method: 'POST',
+      credentials: 'include',
+      body: JSON.stringify({
+        fileId: documentId,
+        pdfType,
+      }),
+    }).then((response) => response.blob())
+      .then((response) => {
+        const pdfFile = new File([response], documentName, { type: 'application/pdf' });
+        console.log(pdfFile);
+
+        const url = window.URL.createObjectURL(response);
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = documentName;
+        document.body.appendChild(a);
+        a.click();
+        a.remove();
+      }).catch((error) => {
+        alert('Error Fetching File');
+      });
+  }
+
+  handleChangeFilePrint(event: any, rowIndex: number) {
```

Comment:
aren't we removing the print?

Reply:
We are not, we are hoping to make it work in the future